fprocjs/src/fproc.js

```diff
@@ -0,0 +1,247 @@
+     * @param {string} command Command to be run
+     * @param {number} customID A custom ID for the command (If left alone, it will automatically set the ID to one that isn't in use)
+     * @param {string} directory The target directory for the command to be ran in (Default is this module's directory)
+     * @param {Object} env The environment variables to be passed down to the process
+     * @returns {Promise<boolean>} Failed or not
+     */
+    async run(command, customID = undefined, directory = __dirname, env = process.env) {
+        if (typeof command !== "string" || command.length <= 0) {
+            throw new FprocError("The command argument must be a string with a content length of at least 1");
+        }
+        if ((typeof customID !== "undefined" && typeof customID !== "number") || customID < 0) {
+            throw new FprocError("The custom ID must be either undefined (meaning unset/auto) or a valid number with a value of at least 0");
+        }
+        if (typeof directory !== "string") {
+            throw new FprocError("The custom directory must be a string");
+        }
+        if (typeof env !== "object") {
+            throw new FprocError("The env must be an object");
+        }
+
+        this.checkErrors();
+        const writer = new HSPB.HolySteamPeerBufferWriter(JSON.stringify(env).length + 100);
+        writer.u8 = headers.Run;
+        writer.string = command;
+        writer.u8 = Number(customID !== undefined);
+        if (customID !== undefined) {
+            writer.u32 = customID;
+        }
+        writer.u32 = Object.keys(env).length;
+        for (const key in env) {
+            writer.string = key;
+            writer.string = env[key].toString();
+        }
+        writer.string = directory;
+        this.fproc.write(writer.buffer);
+        return await this._awaitAsyncRequest(this._errorCheckResponse);
+    }
+
+    /**
+     * Deletes a process from the Fproc daemon
+     * @param {number} id Process ID
+     * @returns {Promise<boolean>} Failed or not
+     */
+    async delete(id) {
+        return await this._idSendRequest(id, headers.Delete);
+    }
+
+    /**
+     * Stops a process in the Fproc daemon
+     * @param {number} id Process ID
+     * @returns {Promise<boolean>} Failed or not
+     */
+    async stop(id) {
+        return await this._idSendRequest(id, headers.Stop);
+    }
+
+    /**
+     * Gets the list of processes running with Fproc
+     * @returns {Promise<FprocList>} Processes info
+     */
+    async list() {
+        this.checkErrors();
+        const writer = new HSPB.HolySteamPeerBufferWriter();
+        writer.u8 = headers.List;
+        this.fproc.write(writer.buffer);
+        return await this._awaitAsyncRequest(this._list);
+    }
+
+    async restart(id) {
+        return await this._idSendRequest(id, headers.Start);
+    }
+
+    /**
+     * Closes the Fproc socket
+     * @returns {void}
+     * @readonly
+     */
+    closeFproc() {
```

Comment:
Perhaps `closeFproc` could be renamed to `closeConnection`.

It would also be good to clarify that the fproc instance is considered destroyed after the connection is closed.

Reply:
Sure